RUFAS/input_manager.py

```diff
@@ -280,6 +289,216 @@ def _filter_input_data_by_metadata(self, input_data: Dict[str, Any],
 
         return filtered_input_data
 
+    def _log_missing_keys_and_keys_with_default_values(
+            self,
+            missing_required_property_keys: List[str],
+            property_keys_with_default_values: List[Tuple[str, Any]]
+    ) -> None:
+        """
+        Logs warnings and errors for missing required properties and properties where default values were applied.
+
+        Notes
+        -----
+        This method aims at providing feedback on the integrity of the data after attempting to fill
+        in missing values based on the schema's default specifications. It serves two main purposes:
+        1. Informing about the absence of required properties that could not be resolved due to
+        a lack of default values.
+        2. Keeping track of those properties for which default values were applied, as this can be
+        useful for debugging and understanding how the data was modified.
+
+        Parameters
+        ----------
+        missing_required_property_keys : List[str]
+            A list of keys for required properties that were not provided in the input data.
+        property_keys_with_default_values : List[Tuple[str, Any]]
+            A list of tuples, each containing a property key and the default value that was applied to it.
+        """
+
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._log_missing_keys_and_keys_with_default_values.__name__,
+        }
+        if missing_required_property_keys:
+            om.add_error("Validation: missing required property keys",
+                         f"Missing required property keys: {', '.join(missing_required_property_keys)}.",
+                         info_map)
+        if property_keys_with_default_values:
+            stringified_default_values = [f"({key} => {value})" for key, value in property_keys_with_default_values]
+            om.add_warning("Validation: missing required property keys",
+                           f"Default values used for required property keys that were missing: "
+                           f"{', '.join(stringified_default_values)}.",
+                           info_map)
+
+    def _add_default_values_to_missing_properties(
```

Comment:
```suggestion
    def _add_default_values_for_missing_inputs(
```
What is missing is not properties but inputs.